### PR TITLE
Delete a few unused functions around stmt.

### PR DESCRIPTION
**Pull request by sandreenko**
d9ebe501ca: Delete incorrect comment.

It was added in 2010 and even when it was added it was wrong as far as I can see.

6673f931b1: Delete unused CopyTestDataToCloneTree.

The only use was deleted in #6102 (3 years ago).

5e2075608f: Delete unused fgSplitBlockAtBeginning.

### CODE REVIEW

File: src/jit/compiler.cpp [hunk 1]
```diff
@@ -6986,123 +6986,6 @@ void Compiler::TransferTestDataToNode(GenTree* from, GenTree* to)
     }
 }
 
-void Compiler::CopyTestDataToCloneTree(GenTree* from, GenTree* to)
```
Comment (mikedn) [inline src/jit/compiler.cpp]: Huh? Is this really unused? Not sure how I missed it, encountered it in my GT_LIST crusade.

**Comment by AndyAyersMS** [comment]
FYI, I am using `fgSplitBlockAtBeginning` in one of my forks. So perhaps keep it around?

File: src/jit/compiler.cpp (hunk 1 above)
Comment (sandreenko) [reply src/jit/compiler.cpp]: I have not found any references since #6102.

**Comment by sandreenko** [comment]
PTAL @dotnet/jit-contrib

**Pull request merged**